lib/models/shows.js

```diff
@@ -0,0 +1,100 @@
+const fetch = require('node-fetch');
+
+async function getTrendingShows() {
+  try {
+    const response = await fetch(
+      `https://api.themoviedb.org/3/trending/tv/week?api_key=${process.env.TMDB_API_KEY}`
+    );
+    const trendingShows = await response.json();
+    const trendingShowsListView = trendingShows.results.map((show) => {
+      const trendingShowListView = {
+        id: show.id,
+        key: show.id,
+        title: show.name,
+        poster: `https://image.tmdb.org/t/p/w154/${show.poster_path}`,
+        rating: Math.floor(show.vote_average),
+      };
+      return trendingShowListView;
+    });
+    return trendingShowsListView;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+}
+
+async function getNewestShows() {
+  const currentDate = new Date();
+  const currentYear = currentDate.getFullYear();
+  const airDateLTE =
+    currentDate.getFullYear() +
+    '-0' +
+    (currentDate.getMonth() + 2) +
+    '-' +
+    currentDate.getDate();
+  const airDateGTE =
+    currentDate.getFullYear() +
+    '-0' +
+    (currentDate.getMonth() - 1) +
+    '-' +
+    currentDate.getDate();
+  try {
+    console.log(airDateGTE);
+    const response = await fetch(
+      `https://api.themoviedb.org/3/discover/tv?api_key=${process.env.TMDB_API_KEY}&language=en-US&sort_by=popularity.desc&air_date.gte=${airDateGTE}&air_date.lte=${airDateLTE}&first_air_date_year=${currentYear}&page=1&vote_average.gte=7&include_null_first_air_dates=false&with_original_language=en`
+    );
+    const newestShows = await response.json();
+    const newestShowsListView = newestShows.results.map((show) => {
+      const newestShowListView = {
+        id: show.id,
+        key: show.id,
+        title: show.name,
+        poster: `https://image.tmdb.org/t/p/w154/${show.poster_path}`,
+        rating: Math.floor(show.vote_average),
+      };
+      return newestShowListView;
+    });
+    return newestShowsListView;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+}
+
+async function getShowDetails(showId) {
+  try {
+    const responses = await Promise.all([
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}?api_key=${process.env.TMDB_API_KEY}&append_to_response=images`),
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}/credits?api_key=${process.env.TMDB_API_KEY}`),
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}/videos?api_key=${process.env.TMDB_API_KEY}`),
+    ]);
+    const allShowDetails = await Promise.all(
+      responses.map((response) => response.json())
+    );
```

Comment:
I think you can remove the first `await Promise.all`

```suggestion
    const responses = [
      fetch(`
    https://api.themoviedb.org/3/tv/${showId}?api_key=${process.env.TMDB_API_KEY}&append_to_response=images`),
      fetch(`
    https://api.themoviedb.org/3/tv/${showId}/credits?api_key=${process.env.TMDB_API_KEY}`),
      fetch(`
    https://api.themoviedb.org/3/tv/${showId}/videos?api_key=${process.env.TMDB_API_KEY}`),
    ];
    const allShowDetails = await Promise.all(
      responses.map((response) => response.json())
    );
```

Reply:
I tried it and it actually marks an error: that response.json() is not a function...